sdk/storage/Azure.Storage.DataMovement.Blobs/src/BlobsStorageResourceProvider.cs

```diff
@@ -222,19 +223,21 @@ public BlobsStorageResourceProvider(GetAzureSasCredential getAzureSasCredentialA
 
         #region Abstract Class Implementation
         /// <inheritdoc/>
-        protected override async Task<StorageResource> FromSourceAsync(DataTransferProperties properties, CancellationToken cancellationToken)
-            => await FromTransferPropertiesAsync(properties, getSource: true, cancellationToken).ConfigureAwait(false);
+        protected override Task<StorageResource> FromSourceAsync(DataTransferProperties properties, CancellationToken cancellationToken)
+            => Task.FromResult(FromTransferProperties(properties, getSource: true, cancellationToken));
 
         /// <inheritdoc/>
-        protected override async Task<StorageResource> FromDestinationAsync(DataTransferProperties properties, CancellationToken cancellationToken)
-            => await FromTransferPropertiesAsync(properties, getSource: false, cancellationToken).ConfigureAwait(false);
+        protected override Task<StorageResource> FromDestinationAsync(DataTransferProperties properties, CancellationToken cancellationToken)
+            => Task.FromResult(FromTransferProperties(properties, getSource: false, cancellationToken));
```

Comment:
So all the rehydrate APIs became sync because they no longer need to read anything for the checkpoint files. They are handed some memory to look at. I left the abstract method signature as async in case we this stuff ever needed to become async again or it needed to be async in a different package. Tbh though, not sure how likely that is with this new design.

Reply:
Consider changing abstract APIs in different PR.